src/convenience/convenience.c

```diff
@@ -360,6 +360,37 @@ int verbose_reset_buffer(SoapySDRDevice *dev)
 	return r;
 }
 
+int verbose_settings(SoapySDRDevice *dev, const char *sdr_settings)
+{
+	char *copied, *cursor, *pair, *equals;
+	int status = 0;
+
+	copied = strdup(sdr_settings);
+	cursor = copied;
+	while ((pair = strsep(&cursor, ",")) != NULL) {
+		char *key, *value;
+
+		equals = strchr(pair, '=');
+		if (equals) {
+			key = pair;
+			*equals = '\0';
+			value = equals + 1;
+		} else {
+			key = pair;
+			value = "";
+		}
```

Comment:
Just FYI, there is a function that does this parsing `SoapySDRKwargs SoapySDRKwargs_fromString(const char *markup);`, its also used for the similar dict of gain names, values.